mcmcfit.py

```diff
@@ -17,12 +17,16 @@
 from model import Model
 import time
 
+from os.path import isfile
+
 sys.settrace
 
 # parallellise with MPIPool
 from emcee.utils import MPIPool
 from six.moves import range
 
+warnings.simplefilter("error", RuntimeWarning)
```

Comment:
I don't think this should be in the production code; it may slow things down significantly. Also, I'm worried about promoting all runtime warnings to errors as a matter of course, since it may cause crashes elsewhere.

If you want this to be merged into the main repo, I'd add a debug option to the mcmc input file, and only do this if debug is enabled. Also, you shouldn't promote warnings to errors everywhere, but should surround just the specific code you want with:

```python
with warnings.catch_warnings():
    warnings.simplefilter("error", RuntimeWarning)
    try:
        .... bad code ....
    except:
       with open('broken_pars.txt', 'a') as f:
           f.write(','.join(self.plist) + '\n')
```

I also think you may have this try: except in the wrong place. Maybe wrap the code in ```LCModel.ln_prob``` instead? That will catch issues in ```ln_prior``` and ```ln_likelihood```. OTOH, if the errors are coming from ln_prior they are easier to fix!

Reply:
I've changed my version to use `with`, but I've kept it in the same place. Other parts of the code use general try:excepts to catch other exceptions, and wrapping this would cause them to misbehave, I think. Errors that don't come from this part should be reported anyway, just not logged, so I'd rather wait for them to present themselves that way and deal with it then.